roles/haproxy/templates/etc/haproxy/haproxy_openstack.cfg

```diff
@@ -154,11 +154,17 @@ backend {{ name }}
   {% else -%}
   option {{ health_check }}
   {% endif -%}
+
   balance {{ balance }}
   {% for host in groups['controller'] -%}
 
   {% if not prefer_primary_backend or hostvars[host][primary_interface]['ipv4']['address'] == primary_ip -%}
+    {% if  name == "keystone" -%}
+    cookie {{ endpoints.keystone.haproxy.oidc_cookie }}
+    server {{ host }} {{ hostvars[host][primary_interface]['ipv4']['address'] }}:{{ clear_port }} cookie check maxconn 40
```

Comment:
The addition of `cookie` in the server line is resulting in this server looking for a cookie named `check`, and then this server doesn't get a httpcheck, so it never sees if the server goes down.